packages/core/src/index.ts

```diff
@@ -10,4 +10,6 @@ export * from './components/avatar';
 export * from './components/text-input';
 export * from './components/theme-provider';
 export * from './components/text';
+export * from './components/h-stack';
+export * from './components/v-stack';
 export * from './components/dialog';
```

Comment:
```suggestion
export { Dialog } from './components/dialog';
```
I suggest exporting as above for the index file and updating package.json as shown in [link](https://github.com/chakra-ui/chakra-ui/blob/08e2000ab0cba25b29eba01a327b6ed146841dfb/packages/react/package.json#L149-L159) for tree-shaking.

What do you think about this approach?

Reply:
@SimYunSup Thank you for your suggestion — I think it's a great idea.

However, after checking the link you shared, it seems that the proposed approach depends on preserving the full project directory structure under the dist/ folder. In the current setup of Vapor UI, all files are bundled due to a known build issue with Vanilla Extract, so it may be difficult to apply this approach right away.

I believe current method is still compatible with RSC, and as long as we avoid using dot notation, tree-shaking should still work reasonably well. What are your thoughts on this?

Also, once we resolve the build issue with Vanilla Extract, we’d be happy to revisit and consider your suggestion more positively.

Reply:
For now, we’ll go ahead and merge this due to business needs.
Feel free to continue the discussion in this PR!